libs/controllers/network/__init__.py

```diff
@@ -1,58 +1,80 @@
-import config as cfg
-
-import _thread
 import asyncio
+import _thread
 import time
+import config as cfg
+
 
 from libs.external.ChannelLogger import logger
 
 
-class Frame:
+class Frame: 
     FRAME_TYPES = {
         'discovery':    0x00,
         'measurement':  0x01,
         'config':       0x02,
         'replication':  0x03,
+        'acknowledgement':0x04,
         'node_joining': 0x06,
         'node_leaving': 0x07,
         'node_alive':   0x08,
         'sync_time':    0x0f,
     }
 
-    def __init__(self, type: int, message: bytes, source_address: int, destination_address: int, ttl=20, rssi=1):
+    def __init__(self, type: int, message: bytes, source_address: int,destination_address: int, ttl=20,frame_num=0,ses_num=0,rssi=1):
         self.type = type
         self.source_address = source_address
         self.destination_address = destination_address
         self.ttl = ttl
+        self.frame_num = frame_num
+        self.ses_num = ses_num        
         self.rssi = rssi
-
         self.data = message
+  
 
     def serialize(self) -> bytes:
+        
         return b''.join([
-            self.type.to_bytes(1, 'big'),
+            self.type.to_bytes(1, 'big'), 
             self.source_address.to_bytes(2, 'big'),
             self.destination_address.to_bytes(2, 'big'),
-            self.data
-        ])
-
+            self.ttl.to_bytes(3, 'big'),
+            self.frame_num.to_bytes(1,'big'),
+            self.ses_num.to_bytes(2,'big'),
+            self.data,
+            #self.crc.to_bytes(2, 'big'),
```

Comment:
Please remove this comment, the crc will be added in a different part of the code